node/pkg/txverifier/suitypes.go

```diff
@@ -375,24 +377,24 @@ func (r SuiTryMultiGetPastObjectsResult) GetTokenChain() (uint16, error) {
 	}
 
 	if !wrapped {
-		return SUI_CHAIN_ID, nil
+		return vaa.ChainIDSui, nil
 	}
 
 	path := "content.fields.value.fields.info.fields.token_chain"
 
-	chain, err := extractFromJsonPath[float64](*r.Details, path)
+	chain, err := extractFromJsonPath[float64](r.Details, path)
 
 	if err != nil {
 		return 0, fmt.Errorf("error in extracting chain: %w", err)
 	}
 
-	return uint16(chain), nil
+	return vaa.ChainID(chain), nil
```

Comment:
Maybe this could be changed to use `KnownChaindIDFromNumber` https://github.com/wormhole-foundation/wormhole/blob/main/sdk/vaa/structs.go#L312 ?
As written this will accept any number and truncate it to uint16. Maybe that's fine in practice, but the check doesn't hurt.